### PR TITLE
CMake: Add CMake support

### PR DESCRIPTION
**Pull request by rajkan01**
- Added CMakeLists.txt file to ntp-client so that it can able build via CMake

@0xc0170 @hugueskamba

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ before_install:
 script:
   # Setup the PR and merge into current master
   - git fetch origin master
-  - git checkout origin/master
+  - git checkout origin/master -b master
```
Comment (0xc0170) [inline .travis.yml]: this still errors in Travis

**Review by 0xc0170 — COMMENTED** [review]
If there is travis already, it should build with CMake

File: .travis.yml (hunk 1 above)
Comment (rajkan01) [reply .travis.yml]: It is failed with or without `-b master`

**Comment by 0xc0170** [comment]
Let's rebase the branch and open new PR (travis should not be there anymore).

**Pull request closed without merging**